mlflow/sklearn/__init__.py

```diff
@@ -654,6 +655,11 @@ def fetch_logged_data(run_id):
     :param silent: If ``True``, suppress all event logs and warnings from MLflow during scikit-learn
                    autologging. If ``False``, show all events and warnings during scikit-learn
                    autologging.
+    :param max_hyper_param_runs: The max number of child mlflow runs to be created for
+                                 hyper parameter search based estimators. If one only cares about
+                                 creating an mlflow child run for the best k results from the
+                                 search, then set max_hyper_param_runs to k. The default value is
+                                 to track the best 5 hyper paramters.
```

Comment:
Can we refactor this phrasing to more succinctly tell the user that the best child runs are selected and also tell them how these runs are selected (i.e. via `rank_test_score` or the first available custom metric)?